test: add comprehensive unit tests for UrlSigner

farion1231 opened the pull request:
- Add tests for signed URL parameter generation
- Add tests for signature verification with various scenarios
- Add tests for security aspects (nonce randomness, signature tampering)
- Add tests for edge cases (empty strings, special characters)
- Ensure proper error handling when SECRET_KEY is not configured


## Checklist

- [ ] This change requires a documentation update, included: [Dify Document](https://github.com/langgenius/dify-docs)
- [x] I understand that this PR may be closed in case there was no previous discussion or issues. (This doesn't apply to typos!)
- [x] I've added a test for each change that was introduced, and I tried as much as possible to make a single atomic change.
- [x] I've updated the documentation accordingly.
- [x] I ran `dev/reformat`(backend) and `cd web && npx lint-staged`(frontend) to appease the lint gods

farion1231 committed:
test: add comprehensive unit tests for UrlSigner

- Add tests for signed URL parameter generation
- Add tests for signature verification with various scenarios
- Add tests for security aspects (nonce randomness, signature tampering)
- Add tests for edge cases (empty strings, special characters)
- Ensure proper error handling when SECRET_KEY is not configured
- Fix pytest.mark.parametrize tuple syntax per PT006
- Apply ruff formatting for consistent code style

farion1231 committed:
refactor: improve test_should_produce_consistent_signatures to use public API

- Remove direct call to private method _sign per Copilot suggestion
- Use public API methods get_signed_url_params and verify instead
- Mock time.time and os.urandom for deterministic testing
- This avoids coupling tests to implementation details

QuantumGhost commented on api/tests/unit_tests/core/helper/test_url_signer.py:
```diff
@@ -0,0 +1,179 @@
+        signed_url = UrlSigner.get_signed_url(base_url, sign_key, prefix)
+
+        # Verify URL contains all required parameters
+        assert signed_url.startswith(base_url)
```
IMO it's better to parse the signed url and assert the query string contains `timestamp`, `nonce` and `sign`.

QuantumGhost reviewed (dismissed):
Other changes LGTM

farion1231 committed:
test: improve URL parsing in test_should_generate_complete_signed_url

- Replace string contains checks with proper URL parsing using urlparse and parse_qs
- Verify base URL structure is preserved after adding query parameters
- Validate query parameters exist, have single values, and are non-empty
- More robust testing that handles parameter order changes

Per QuantumGhost's review suggestion

QuantumGhost reviewed (approved):
LGTM.

Thank you for your contribution.
